lib/publiccloud/instance.pm

```diff
@@ -119,6 +122,90 @@ sub retry_ssh_command {
     die "Waiting for Godot: " . $cmd;
 }
 
+# Auxilliary function to prepare the ssh command that runs any command on the PC instance
+sub _prepare_ssh_cmd {
+    my ($self, %args) = @_;
+    die('No command defined') unless ($args{cmd});
+    $args{ssh_opts} //= $self->ssh_opts() . " -i '" . $self->ssh_key . "'";
+    $args{username} //= $self->username();
+    $args{timeout} //= SSH_TIMEOUT;
+
+    my $cmd = $args{cmd};
+    unless ($args{no_quote}) {
+        $cmd =~ s/'/\'/g;    # Espace ' character
+        $cmd = "\$'$cmd'";
```

Comment:
sorry but my perl/bash knowledge not allowing me to understand purpose of this line , can you elaborate ?